src/slices/main/fab/Fab.js

```diff
@@ -79,13 +143,22 @@ class Fab extends Component {
     };
 }
 
+Fab.defaultProps = {
+    in: 'bounce-in',
+    forwardedRef: noop,
+    onAnimationEnd: noop,
+};
+
 Fab.propTypes = {
+    in: PropTypes.oneOf(['bounce-in', 'scale-up', 'scale-down', 'bounce-out']),
+    forwardedRef: PropTypes.func,
+    scaleAmount: PropTypes.number,
     authenticated: PropTypes.bool,
     className: PropTypes.string,
     onOpen: PropTypes.func.isRequired,
     onAuthenticationOpen: PropTypes.func.isRequired,
     onAuthenticationClose: PropTypes.func.isRequired,
-    onAnimationEnd: PropTypes.func.isRequired,
+    onAnimationEnd: PropTypes.func,
 };
 
-export default Fab;
+export default React.forwardRef((props, ref) => <Fab { ...props } forwardedRef={ ref } />);
```

Comment:
It seems strange that passing a `ref` property to the Fab would give the DOM element and not the Fab component. Perhaps, instead of this "magic":

1. Add a `getDOMNode` into the `Fab` and use/call it to obtain the node?
2. Use `findDOMNode` on the `Fab` ref to obtain the DOMNode?
3. Explicitly use a `rootRef` prop or similar name

**Note** that the same applies for the Sidebar component.

Reply:
I see we're using `findDOMNode` in some components. For the sake of consistency, should we go for `findDOMNode`?

Reply:
I’m fine with that 👍

Reply:
Done!